cluster-http/src/main/scala/akka/cluster/http/ClusterHttpApi.scala

```diff
@@ -0,0 +1,145 @@
+      None
+  }
+}
+
+trait ClusterHttpApiJsonProtocol extends SprayJsonSupport with DefaultJsonProtocol {
+  implicit val clusterUnreachableMemberFormat = jsonFormat2(ClusterUnreachableMember)
+  implicit val clusterMemberFormat = jsonFormat4(ClusterMember)
+  implicit val clusterMembersFormat = jsonFormat3(ClusterMembers)
+  implicit val clusterMemberMessageFormat = jsonFormat1(ClusterHttpApiMessage)
+}
+
+trait ClusterHttpApiHelper extends ClusterHttpApiJsonProtocol {
+  def memberToClusterMember(m: Member): ClusterMember = {
+    ClusterMember(s"${m.uniqueAddress.address}", s"${m.uniqueAddress.uid}", s"${m.status}", m.roles)
+  }
+}
+
+object ClusterHttpRoute extends ClusterHttpApiHelper {
+
+  private def routeGetMembers(cluster: Cluster) = {
+    get {
+      complete {
+        val members = cluster.readView.state.members.map(memberToClusterMember)
+
+        val unreachable = cluster.readView.reachability.observersGroupedByUnreachable.toSeq.sortBy(_._1).map {
+          case (subject, observers) ⇒
+            ClusterUnreachableMember(s"${subject.address}", observers.toSeq.sorted.map(m ⇒ s"${m.address}"))
+        }
+
+        ClusterMembers(s"${cluster.readView.selfAddress}", members, unreachable)
+      }
+    }
+  }
+
+  private def routePostMembers(cluster: Cluster) = {
+    post {
+      formField('address) { address ⇒
+        complete {
+          cluster.join(AddressFromURIString(address))
+          ClusterHttpApiMessage("Operation executed")
+        }
+      }
+    }
+  }
+
+  private def routeGetMember(cluster: Cluster, member: String) =
+    get {
+      complete {
+        cluster.readView.members.find(m ⇒ s"${m.uniqueAddress.address}" == member).map(memberToClusterMember)
+      }
+    }
+
+  private def routeDeleteMember(cluster: Cluster, member: String) =
+    delete {
+      cluster.readView.members.find(m ⇒ s"${m.uniqueAddress.address}" == member) match {
+        case Some(m) =>
+          cluster.leave(m.uniqueAddress.address)
+          complete(ClusterHttpApiMessage("Operation executed"))
+        case None =>
+          complete(StatusCodes.NotFound -> ClusterHttpApiMessage("Member not found"))
+      }
+    }
+
+  private def routePutMember(cluster: Cluster, member: String) =
+    put {
+      formField('operation) { operation =>
+        cluster.readView.members.find(m ⇒ s"${m.uniqueAddress.address}" == member) match {
+          case Some(m) =>
+            ClusterHttpApiOperation.find(operation.toUpperCase) match {
+              case Some(ClusterHttpApiOperation.DOWN) =>
+                cluster.down(m.uniqueAddress.address)
+                complete(ClusterHttpApiMessage("Operation executed"))
+              case Some(ClusterHttpApiOperation.LEAVE) =>
+                cluster.leave(m.uniqueAddress.address)
+                complete(ClusterHttpApiMessage("Operation executed"))
+              case _ =>
+                complete(StatusCodes.NotFound -> ClusterHttpApiMessage("Operation not supported"))
```

Comment:
Let's use `400 Bad Request` instead here, it's a client error here, not a not found.